### ISSUE TITLE
Test failing in `beforeEach` is not reported as failed

### ISSUE THREAD

**Issue by AnnikaNissen** [sullenor/wdio-teamcity-reporter#21]
When having a test, that fails in the `beforeEach` block, it is not reported as failed by the TeamCity reporter. The TeamCity UI shows the test as "run & passed", which is not correct and very misleading.

Here is a test to reproduce and the output I get. The test is reported correctly by the spec reporter.

Test
```typescript
describe('Before each', function() {
    beforeEach(function() {
        browser.url('https://www.google.com');
        throw new Error("Simulate test failing in beforeEach");
    });

    it('test should fail with error in "beforeEach"', function() {
        expect(browser).toHaveUrlContaining('google');
    });
});
```

TeamCity reporter output:
```
[0-0] ##teamcity[testSuiteStarted name='Before each' flowId='3e837eaa83a635949c0abb4a8b45d276/0-0']
[0-0] ##teamcity[testStarted name='test should fail with error in "beforeEach"' captureStandardOutput='false' flowId='3e837eaa83a635949c0abb4a8b45d276/0-0']
[0-0] ##teamcity[testSuiteFinished name='Before each' flowId='3e837eaa83a635949c0abb4a8b45d276/0-0']
```

Spec reporter output:
```
[chrome 86.0.4240.183 mac os x #0-0] Spec: ../test-desktop/specs/before-each.spec.ts
[chrome 86.0.4240.183 mac os x #0-0] Running: chrome (v86.0.4240.183) on mac os x
[chrome 86.0.4240.183 mac os x #0-0] Session ID: 3e837eaa83a635949c0abb4a8b45d276
[chrome 86.0.4240.183 mac os x #0-0]
[chrome 86.0.4240.183 mac os x #0-0] Before each
[chrome 86.0.4240.183 mac os x #0-0]    ? test should fail with error in "beforeEach"
[chrome 86.0.4240.183 mac os x #0-0]    ✖ "before each" hook for "test should fail with error in "beforeEach""
[chrome 86.0.4240.183 mac os x #0-0]
[chrome 86.0.4240.183 mac os x #0-0] 1 failing (2.9s)
[chrome 86.0.4240.183 mac os x #0-0]
[chrome 86.0.4240.183 mac os x #0-0] 1) Before each "before each" hook for "test should fail with error in "beforeEach""
[chrome 86.0.4240.183 mac os x #0-0] Simulate test failing in beforeEach
[chrome 86.0.4240.183 mac os x #0-0] Error: Simulate test failing in beforeEach
[chrome 86.0.4240.183 mac os x #0-0]     at Context.<anonymous> (../test-desktop/specs/before-each.spec.ts:4:15)
```

**Comment by AnnikaNissen** [comment]
I did some investigations on this today and found a solution. Will try to provide a PR for it within the next couple of days.

**Comment by sullenor** [comment]
Hey @AnnikaNissen, thanks for flagging this one. In case you'll find any — feel free to reach me out :)

**Comment by AnnikaNissen** [comment]
@sullenor I finally managed to provide a PR for this: https://github.com/sullenor/wdio-teamcity-reporter/pull/22 
At the moment I'm waiting for Travis CI to see if it works. I'm happy to get your feedback on the PR :-)

**Comment by sullenor** [comment]
Thanks @AnnikaNissen

I merged PR and publish version. The change should be available starting 2.3.0 version.

Closing the issue since I believe it's resolved for now :)
